metrics_v2/handler/main.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import itertools
+import os
+import typing
+import uuid
+
+from absl import logging
+import google.auth
+import numpy as np
+import sendgrid
+from sendgrid.helpers import mail
+
+import alerts
+import bigquery_client
+import collectors
+import utils
+import metrics_pb2
+
+try:
+  DATASET = os.environ['BQ_DATASET']
+except KeyError:
+  raise KeyError('Must set $BQ_DATASET env var.')
+
+SEND_EMAIL_ALERTS = os.getenv('SEND_EMAIL_ALERTS', None)
+PROJECT = os.getenv('GCP_PROJECT', None)
+
+SOURCE_TO_COLLECTOR = {
+  'literals': collectors.literal.LiteralCollector,
+  'perfzero': collectors.perfzero.PerfZeroCollector,
+  'tensorboard': collectors.tensorboard.TensorBoardCollector,
+}
+
+_SENDGRID_API_SECRET_NAME = 'sendgrid-api-key'
```

Comment:
maybe this is where the other comment about secrets was meant to be?

Reply:
Yes, I moved these constants from alerts and missed the comment.